packages/core/src/TextSplitter.ts

```diff
@@ -1,5 +1,5 @@
+import { EOL } from 'node:os'
```

Comment:
@himself65 does this run with Deno and/or Bun? Do you know a website like caniuse.com for this?

Reply:
first, this is node-only, but Deno supports `node:xxx` URL schema now

second, caniuse is for Web API

Reply:
I think there is no such website?

Reply:
I also couldn't find such a website.

How about we put this platform-specific code in an own folder, like `packages/core/src/platforms/node`, and we import `EOL` from there?

In the long run, we could put all platform-specific code there, e.g. we're having this in `FileSystem.ts`:
```
export function getNodeFS(): CompleteFileSystem {
  const fs = require("fs/promises");
  return fs;
}
```
More and more platforms are coming out, and keeping this code in a dedicated place is good.
@yisding, what are your thoughts?